apps/explorer/src/app/routes/assets/index.tsx

```diff
@@ -1,63 +1,38 @@
-import { gql, useQuery } from '@apollo/client';
 import { t } from '@vegaprotocol/react-helpers';
 import React from 'react';
 import { RouteTitle } from '../../components/route-title';
 import { SubHeading } from '../../components/sub-heading';
 import { SyntaxHighlighter } from '@vegaprotocol/ui-toolkit';
-import type {
-  AssetsQuery,
-  AssetsQuery_assetsConnection_edges_node,
-} from './__generated__/AssetsQuery';
-
-export const ASSETS_QUERY = gql`
-  query AssetsQuery {
-    assetsConnection {
-      edges {
-        node {
-          id
-          name
-          symbol
-          decimals
-          source {
-            ... on ERC20 {
-              contractAddress
-            }
-            ... on BuiltinAsset {
-              maxFaucetAmountMint
-            }
-          }
-          infrastructureFeeAccount {
-            type
-            balance
-            market {
-              id
-            }
-          }
-        }
-      }
-    }
-  }
-`;
+import { useExplorerAssetsQuery } from './__generated__/assets';
 
 const Assets = () => {
-  const { data } = useQuery<AssetsQuery>(ASSETS_QUERY);
+  const { data } = useExplorerAssetsQuery();
 
-  const assets =
-    data?.assetsConnection?.edges
-      ?.filter((e) => e && e?.node)
-      .map((e) => e?.node as AssetsQuery_assetsConnection_edges_node) || [];
+  const assets = data?.assetsConnection?.edges?.map((n) => {
```

Comment:
comment (non-blocking): added a convenience function to grab nodes from these paginated structures, it lives in the react helpers:

```
import {getNodes} from '@vegaprotocol/react-helpers' 

const nodes = getNodes(data?.assetsConnection)
```

not sure if it will work everywhere, might need to take a type parameter for the actual node type, which is usually not generated out of the box, unless you define it as a fragment. probably not super important, just in the future if you get bored with mapping over edges, feel free to give it a go :)

Reply:
Thanks - I will use that in future, but given how many files this PR touches I'm not going to make the change in this PR. Ticketing for later.

Reply:
makes sense